src/js/common/components/RequiredLoginView.jsx

```diff
@@ -0,0 +1,152 @@
+import React from 'react';
+import { connect } from 'react-redux';
+
+import { commonStore } from '../store';
+
+import environment from '../helpers/environment.js';
+import { updateLoggedInStatus, updateLogInUrl, updateProfileField } from '../actions';
+
+class RequiredLoginView extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleLogin = this.handleLogin.bind(this);
+    this.handleVerify = this.handleVerify.bind(this);
+    this.setMyToken = this.setMyToken.bind(this);
+    this.getUserData = this.getUserData.bind(this);
+  }
+
+  componentDidMount() {
+    if (localStorage.length > 0) {
+      this.props.onUpdateLoggedInStatus(true);
+      this.getUserData();
+    } else {
+      this.props.onUpdateLoggedInStatus(false);
+    }
+
+    window.addEventListener('message', this.setMyToken);
+  }
+
+  setMyToken() {
+    if (event.data === localStorage.userToken) {
+      this.getUserData();
+    }
+  }
+
+  getUserData() {
+    fetch(`${environment.API_URL}/v0/user`, {
+      method: 'GET',
+      headers: new Headers({
+        Authorization: `Token token=${localStorage.userToken}`
+      })
+    }).then(response => {
+      return response.json();
+    }).then(json => {
+      // console.log(json);
+      const userData = json.data.attributes.profile;
+      this.props.onUpdateProfile('accountType', userData.loa.current);
+      this.props.onUpdateProfile('email', userData.email);
+      this.props.onUpdateProfile('userFullName.first', userData.first_name);
+      this.props.onUpdateProfile('userFullName.middle', userData.middle_name);
+      this.props.onUpdateProfile('userFullName.last', userData.last_name);
+      // this.props.onUpdateProfile('userFullName.suffix', userData.first_name);
+      this.props.onUpdateProfile('gender', userData.gender);
+      this.props.onUpdateProfile('dob', userData.birth_date);
+      this.props.onUpdateLoggedInStatus(true);
+    });
+  }
+
+  handleLogin() {
+    const myStore = commonStore.getState();
+    const login = myStore.login;
+    const myLoginUrl = login.loginUrl.first;
+    const receiver = window.open(myLoginUrl, '_blank', 'resizable=yes,top=50,left=500,width=500,height=750');
+    receiver.focus();
+  }
+
+  handleVerify() {
+    const myStore = commonStore.getState();
+    const login = myStore.login;
+    const myVerifyUrl = login.loginUrl.third;
```

Comment:
It's safe to assume this order will stay the same? (i.e. pulling third will always work?)